src/features/bivariate_color_manager/atoms/bivariateColorManagerResource.ts

```diff
@@ -99,39 +103,46 @@ export const bivariateColorManagerResourceAtom = createResourceAtom(
         return acc;
       }, {});
 
-      const axisNominatorQualityMap = axis.reduce<AxisNominatorQualityMap>(
-        (acc, axis) => {
-          if (!axis.quality) return acc;
-
-          const [numerator, denominator] = axis.quotient;
-          if (!acc[numerator]) {
-            acc[numerator] = [denominator, axis.quality];
-          } else {
-            const prevAxisQualityByNumerator = acc[numerator][1];
-            if (prevAxisQualityByNumerator < axis.quality) {
-              acc[numerator] = [denominator, axis.quality];
-            }
+      const axisNominatorInfo = axis.reduce<AxisNominatorInfo>((acc, axis) => {
+        if (!axis.quality || axis.quality < 0.5) return acc; // we don't need low quality axises
+
+        const [numerator, denominator] = axis.quotient;
+        if (!acc[numerator]) {
+          acc[numerator] = {
+            denominator,
+            quality: axis.quality,
+            layersCount: 1,
+          };
+        } else {
+          acc[numerator].layersCount++;
+
+          const prevAxisQualityByNumerator = acc[numerator].quality;
+          if (prevAxisQualityByNumerator < axis.quality) {
+            acc[numerator] = {
+              ...acc[numerator],
+              denominator,
+              quality: axis.quality,
+            };
           }
```

Comment:
So, we looking for most quality axis here?

Reply:
most quality denominator
it looks like it's better to add comment, isn't it?  🤔

Reply:
I've decided to change naming to be clear

acc[numerator] = {
            mostQualityDenominator: denominator,
            quality: axis.quality,
            layersCount: 1,
          };
          
i think it's clear that for numerator you get mostQualityDenominator, quality, layersCount
does it look ok? @Akiyamka

Reply:
yep, looks better